src/App.js

```diff
@@ -190,6 +196,12 @@ const App = ({ locationId }) => {
                     navigate("/");
                   } else {
                     navigate(`/locations/${location.id}`);
+                    setTransitionDuration(animationMS);
```

Comment:
Why does this need to be a hook and set every time vs. just set directly as part of the `ReactMapGL` props?

Reply:
It needs to be set every time because the value when we aren't panning to a selected pin should be the animation duration but for all other gestures (pan, zoom) it needs to be 0. `ReactMapsGL` doesn't have a very robust way of handling transitions. 

So we're toggling the duration between 0 and the animation length. As soon as the transition to the selected marker in the centre finishes, we reset the value to 0 so the user can pan/zoom normally (otherwise the `transitionDuration` value applies to each from of those gestures... not good!)

So for each time they select a marker, the value will be 0 so we set it to 500 then update the viewport. Then on transition end, if the value is not 0, set `transitionDuration` back to 0.

Reply:
Ah, makes sense! Thanks for the explanation.